python-lib/api_formatting.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+import logging
+from typing import AnyStr, Dict, List
+from enum import Enum
+
+import pandas as pd
+import boto3
+from boto3.exceptions import Boto3Error
+from botocore.exceptions import BotoCoreError, ClientError
+
+from plugin_io_utils import (
+    API_COLUMN_NAMES_DESCRIPTION_DICT,
+    ErrorHandlingEnum,
+    build_unique_column_names,
+    generate_unique,
+    safe_json_loads,
+    move_api_columns_to_end,
+)
+
+
+# ==============================================================================
+# CONSTANT DEFINITION
+# ==============================================================================
+
+API_EXCEPTIONS = (Boto3Error, BotoCoreError, ClientError)
+
+API_SUPPORT_BATCH = True
+BATCH_RESULT_KEY = "ResultList"
+BATCH_ERROR_KEY = "ErrorList"
+BATCH_INDEX_KEY = "Index"
+BATCH_ERROR_MESSAGE_KEY = "ErrorMessage"
+BATCH_ERROR_TYPE_KEY = "ErrorCode"
+
+VERBOSE = False
+
+
+class EntityTypeEnum(Enum):
+    COMMERCIAL_ITEM = "Commercial item"
+    DATE = "Date"
+    EVENT = "Event"
+    LOCATION = "Location"
+    ORGANIZATION = "Organization"
+    OTHER = "Other"
+    PERSON = "Person"
+    QUANTITY = "Quantity"
+    TITLE = "Title"
+
+
+# ==============================================================================
+# CLASS AND FUNCTION DEFINITION
+# ==============================================================================
+
+
+def get_client(api_configuration_preset, service_name: AnyStr):
```

Comment:
I am not sure get_client should there. Must be move to plugin_io_utils

Reply:
@MayeulRousselet the way I've structured the code, plugin_io_utils and api_formatting are always the same across plugins. The only library file which changes is "api_formatting" across plugins. 
Maybe it's a file naming issue? I could rename it to dku_amazon_comprehend.py?

Reply:
For me the method get_client is not about api_formatting, so indeed it is a naming issue

Reply:
agreed, so what about api_specific_utils? Or do you prefer a name with the cloud name (cons: it would make api_parallelizer not generic)?

Reply:
or should I move it to a dedicated file? in which case it will be a short file